tpm/key.go

```diff
@@ -104,6 +105,15 @@ func (k *Key) MarshalJSON() ([]byte, error) {
 	return json.Marshal(o)
 }
 
+// ToTSS2 gets the public and private blobs and returns a [*tss2.TPMKey].
+func (k *Key) ToTSS2(ctx context.Context) (*tss2.TPMKey, error) {
```

Comment:
Somewhat related to my `tpm.TPM` remark: maybe a `ToTSS2Signer` also makes sense to add on `Key` and `AK`? I think it can get the reference to the `tpm.TPM` in one go that way?

Reply:
Are you saying to add a `ToTSS2Signer` or change this name to `ToTSS2Signer`? 

We need a method to be able to transform a Key/AK to a `*tss2.TPMKey`, that type by itself is not a signer. And adding a `ToTSS2Signer()` doesn't really make a lot of sense, we already have `GetSigner()`:

```
func (t *TPM) GetSigner(ctx context.Context, name string) (csigner crypto.Signer, err error) 
```

Reply:
`GetSigner` only works with AKs and Keys natively stored behind the TPM facade and identified by `name`, though. 

I was thinking about adding an additional `ToTSS2Signer` method, which would function similar to the (now) `CreateTSS2Signer`, but would be defined on the `Key` and `AK`, and then attach the reference to the `tpm.TPM` in that method immediately. Then you can get the signer without having to call `CreateTSS2Signer` explicitly.